lib/wallet/http.js

```diff
@@ -2157,6 +1727,38 @@ class HTTP extends Server {
       return null;
     });
   }
+
+  /**
+   * Registers a new http handler at provided path
+   * the handler method is passed regular req, res along
+   * with a context object that holds vallet specific
+   * objects that might be used to extend wallet plugin
+   * functionality
+   *
+   * @param {*} httpMethod string value of one of
+   * 'post', 'get', 'put', 'patch', 'delete'
+   * @param {*} path location handler is listening
+   * @param {*} handler http handler that will handle requests
+   * arriving to `path` registered
+   * @param {Object} context context
+   * @returns {undefined}
+   */
+   registerRoute(httpMethod, path, handler, context) {
```

Comment:
Nice, would be good to have some way of detecting duplicate paths to avoid bugs but not blocking